addons/web_editor/static/src/js/editor/odoo-editor/test/spec/editor.test.js

```diff
@@ -3517,6 +3517,29 @@ X[]
                     contentAfter: `[]<br>`,
                 });
             });
+            describe('Across multiple list types', () => {
+                it ('should merge a list item from the first list into the second list', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: `<ul><li>ab</li><li>[cd</li></ul><ol><li>ef]</li><li>gh</li></ol>`,
+                        stepFunction: deleteBackward,
+                        contentAfter: `<ul><li>ab</li></ul><ol><li>[]<br></li><li>gh</li></ol>`,
+                    });
+                });
+                it ('should not merge a list item from the first list into the second list', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: `<ul><li>ab</li><li>c[d</li></ul><ol><li>e]f</li><li>gh</li></ol>`,
+                        stepFunction: deleteBackward,
+                        contentAfter: `<ul><li>ab</li><li>c[]f</li></ul><ol><li>gh</li></ol>`,
+                    });
```

Comment:
I could not find this test in html_editor?

Reply:
@dmo-odoo It's similar to this case, which is why I haven't added it.
https://github.com/odoo/odoo/blob/18.0/addons/html_editor/static/tests/list/delete_backward.test.js#L2303